diff --git a/MsCorePkg/Library/PlatformBootManagerLib/BdsPlatform.c b/MsCorePkg/Library/PlatformBootManagerLib/BdsPlatform.c
index e4735d5d..66efac70 100644
--- a/MsCorePkg/Library/PlatformBootManagerLib/BdsPlatform.c
+++ b/MsCorePkg/Library/PlatformBootManagerLib/BdsPlatform.c
@@ -8,6 +8,8 @@ SPDX-License-Identifier: BSD-2-Clause-Patent
 
 #include "BdsPlatform.h"
 
+#include <Library/BootGraphicsLib.h>
+
 static EFI_BOOT_MODE                 mBootMode;
 static EFI_DEVICE_PATH_PROTOCOL      **mPlatformConnectSequence;
 static USB_CLASS_FORMAT_DEVICE_PATH  mUsbClassKeyboardDevicePath = {
@@ -471,5 +473,16 @@ PlatformBootManagerWaitCallback (
   UINT16  TimeoutRemain
   )
 {
-  return;
+  EFI_STATUS Status;
+
+  if (!TimeoutRemain) {
+    // Clear Screen
+    gST->ConOut->ClearScreen (gST->ConOut);
+
+    // Display Boot Logo
+    Status = DisplayBootGraphic (BG_SYSTEM_LOGO);
+    if (EFI_ERROR (Status)) {
+      DEBUG ((EFI_D_ERROR, "%a: Failed to Display Boot Logo! Status = %r\n", __FUNCTION__, Status));
+    }
+  }
 }
diff --git a/MsCorePkg/Library/PlatformBootManagerLib/PlatformBootManagerLib.inf b/MsCorePkg/Library/PlatformBootManagerLib/PlatformBootManagerLib.inf
index e8b4faf8..ff62fda1 100644
--- a/MsCorePkg/Library/PlatformBootManagerLib/PlatformBootManagerLib.inf
+++ b/MsCorePkg/Library/PlatformBootManagerLib/PlatformBootManagerLib.inf
@@ -34,6 +34,7 @@ CONSTRUCTOR                    = PlatformBootManagerEntry
   MdeModulePkg/MdeModulePkg.dec
   MsCorePkg/MsCorePkg.dec
   SecurityPkg/SecurityPkg.dec
+  MsGraphicsPkg/MsGraphicsPkg.dec
 
 [LibraryClasses]
   BaseLib
@@ -51,6 +52,7 @@ CONSTRUCTOR                    = PlatformBootManagerEntry
   UefiBootServicesTableLib
   UefiLib
   UefiRuntimeServicesTableLib
+  BootGraphicsLib
 
 [Protocols]
   gEfiPciRootBridgeIoProtocolGuid     ## CONSUMES
