diff --git a/MdeModulePkg/Application/BootManagerMenuApp/BootManagerMenu.c b/MdeModulePkg/Application/BootManagerMenuApp/BootManagerMenu.c
index ef19319614..df8d1ac040 100644
--- a/MdeModulePkg/Application/BootManagerMenuApp/BootManagerMenu.c
+++ b/MdeModulePkg/Application/BootManagerMenuApp/BootManagerMenu.c
@@ -6,6 +6,8 @@ SPDX-License-Identifier: BSD-2-Clause-Patent
 
 **/
 
+#include <Library/UefiRuntimeServicesTableLib.h>
+
 #include "BootManagerMenu.h"
 
 EFI_HII_HANDLE  gStringPackHandle;
@@ -344,9 +346,9 @@ IgnoreBootOption (
   }
 
   //
-  // Ignore the hidden/inactive boot option.
+  // Ignore "UEFI Misc Device" boot options.
   //
-  if (((BootOption->Attributes & LOAD_OPTION_HIDDEN) != 0) || ((BootOption->Attributes & LOAD_OPTION_ACTIVE) == 0)) {
+  if (BootOption->Description != NULL && StrnCmp (BootOption->Description, L"UEFI Misc Device", 16) == 0) {
     return TRUE;
   }
 
@@ -379,7 +381,7 @@ InitializeBootMenuData (
   }
 
   BootMenuData->TitleToken[0] = STRING_TOKEN (STR_BOOT_POPUP_MENU_TITLE_STRING);
-  BootMenuData->PtrTokens     = AllocateZeroPool (BootOptionCount * sizeof (EFI_STRING_ID));
+  BootMenuData->PtrTokens     = AllocateZeroPool ((BootOptionCount + 2) * sizeof (EFI_STRING_ID));
   ASSERT (BootMenuData->PtrTokens != NULL);
 
   //
@@ -399,10 +401,19 @@ InitializeBootMenuData (
                                             );
   }
 
+  //
+  // Add Power Off Option
+  //
+  BootMenuData->PtrTokens[StrIndex++] = HiiSetString (gStringPackHandle, 0, L"Power Off", NULL);
+
+  //
+  // Add Exit Option
+  //
+  BootMenuData->PtrTokens[StrIndex++] = HiiSetString (gStringPackHandle, 0, L"Exit", NULL);
+
   BootMenuData->ItemCount    = StrIndex;
   BootMenuData->HelpToken[0] = STRING_TOKEN (STR_BOOT_POPUP_MENU_HELP1_STRING);
   BootMenuData->HelpToken[1] = STRING_TOKEN (STR_BOOT_POPUP_MENU_HELP2_STRING);
-  BootMenuData->HelpToken[2] = STRING_TOKEN (STR_BOOT_POPUP_MENU_HELP3_STRING);
   InitializeBootMenuScreen (BootMenuData);
   BootMenuData->SelectItem = 0;
   return EFI_SUCCESS;
@@ -468,7 +479,7 @@ BootMenuSelectItem (
       BootMenuData->ScrollBarControl.LastItem  = WantSelectItem;
     }
 
-    gST->ConOut->SetAttribute (gST->ConOut, EFI_WHITE | EFI_BACKGROUND_BLUE);
+    gST->ConOut->SetAttribute (gST->ConOut, EFI_WHITE | EFI_BACKGROUND_BLACK);
     FirstItem   = BootMenuData->ScrollBarControl.FirstItem;
     LastItem    = BootMenuData->ScrollBarControl.LastItem;
     TopShadeNum = 0;
@@ -539,7 +550,7 @@ BootMenuSelectItem (
   //
   FirstItem = BootMenuData->ScrollBarControl.FirstItem;
   if ((WantSelectItem != BootMenuData->SelectItem) && !RePaintItems) {
-    gST->ConOut->SetAttribute (gST->ConOut, EFI_WHITE | EFI_BACKGROUND_BLUE);
+    gST->ConOut->SetAttribute (gST->ConOut, EFI_WHITE | EFI_BACKGROUND_BLACK);
     String   = HiiGetString (gStringPackHandle, BootMenuData->PtrTokens[BootMenuData->SelectItem], NULL);
     PrintCol = StartCol  + 1;
     PrintRow = StartRow + 3 + BootMenuData->SelectItem - FirstItem;
@@ -550,7 +561,7 @@ BootMenuSelectItem (
   //
   // Print want to select item
   //
-  gST->ConOut->SetAttribute (gST->ConOut, EFI_WHITE | EFI_BACKGROUND_BLACK);
+  gST->ConOut->SetAttribute (gST->ConOut, EFI_WHITE | EFI_BACKGROUND_LIGHTGRAY);
   String   = HiiGetString (gStringPackHandle, BootMenuData->PtrTokens[WantSelectItem], NULL);
   PrintCol = StartCol  + 1;
   PrintRow = StartRow + TITLE_TOKEN_COUNT + 2 + WantSelectItem - FirstItem;
@@ -589,7 +600,7 @@ DrawBootPopupMenu (
   gST->ConOut->ClearScreen (gST->ConOut);
 
   SavedAttribute = gST->ConOut->Mode->Attribute;
-  gST->ConOut->SetAttribute (gST->ConOut, EFI_WHITE | EFI_BACKGROUND_BLUE);
+  gST->ConOut->SetAttribute (gST->ConOut, EFI_WHITE | EFI_BACKGROUND_BLACK);
   Width              = BootMenuData->MenuScreen.Width;
   StartCol           = BootMenuData->MenuScreen.StartCol;
   StartRow           = BootMenuData->MenuScreen.StartRow;
@@ -754,6 +765,16 @@ BootFromSelectOption (
       break;
     }
   }
+
+  // Power Off
+  if (SelectItem == ItemNum) {
+    gRT->ResetSystem (EfiResetShutdown, EFI_SUCCESS, 0, NULL);
+  }
+
+  // Exit
+  if (SelectItem == ItemNum + 1) {
+    gBS->Exit (gImageHandle, EFI_SUCCESS, 0, NULL);
+  }
 }
 
 /**
diff --git a/MdeModulePkg/Application/BootManagerMenuApp/BootManagerMenu.h b/MdeModulePkg/Application/BootManagerMenuApp/BootManagerMenu.h
index 876237df18..2aa50e256e 100644
--- a/MdeModulePkg/Application/BootManagerMenuApp/BootManagerMenu.h
+++ b/MdeModulePkg/Application/BootManagerMenuApp/BootManagerMenu.h
@@ -23,7 +23,7 @@ SPDX-License-Identifier: BSD-2-Clause-Patent
 #include <Protocol/BootLogo.h>
 
 #define TITLE_TOKEN_COUNT  1
-#define HELP_TOKEN_COUNT   3
+#define HELP_TOKEN_COUNT   2
 
 typedef struct _BOOT_MENU_SCREEN {
   UINTN    StartCol;
