diff --git a/ArmPkg/Drivers/CpuDxe/AArch64/Mmu.c b/ArmPkg/Drivers/CpuDxe/AArch64/Mmu.c
index 36c996be47..602ac125a7 100644
--- a/ArmPkg/Drivers/CpuDxe/AArch64/Mmu.c
+++ b/ArmPkg/Drivers/CpuDxe/AArch64/Mmu.c
@@ -68,14 +68,6 @@ PageAttributeToGcdAttribute (
       GcdAttributes = EFI_MEMORY_WB;
       break;
     default:
-      DEBUG ((
-        DEBUG_ERROR,
-        "PageAttributeToGcdAttribute: PageAttributes:0x%lX not supported.\n",
-        PageAttributes
-        ));
-      ASSERT (0);
-      // The Global Coherency Domain (GCD) value is defined as a bit set.
-      // Returning 0 means no attribute has been set.
       GcdAttributes = 0;
   }
 
