diff --git a/ArmPkg/Library/ArmGenericTimerPhyCounterLib/ArmGenericTimerPhyCounterLib.c b/ArmPkg/Library/ArmGenericTimerPhyCounterLib/ArmGenericTimerPhyCounterLib.c
index fbbecdc528..9378c27017 100644
--- a/ArmPkg/Library/ArmGenericTimerPhyCounterLib/ArmGenericTimerPhyCounterLib.c
+++ b/ArmPkg/Library/ArmGenericTimerPhyCounterLib/ArmGenericTimerPhyCounterLib.c
@@ -59,7 +59,14 @@ ArmGenericTimerGetTimerFreq (
   VOID
   )
 {
-  return ArmReadCntFrq ();
+  UINTN Freq;
+
+  Freq = ArmReadCntFrq ();
+  if (!Freq) {
+    Freq = FixedPcdGet64 (PcdTimerFreqOverwrite);
+  }
+
+  return Freq;
 }
 
 UINTN
diff --git a/ArmPkg/Library/ArmGenericTimerPhyCounterLib/ArmGenericTimerPhyCounterLib.inf b/ArmPkg/Library/ArmGenericTimerPhyCounterLib/ArmGenericTimerPhyCounterLib.inf
index c37f2729b1..0462650680 100644
--- a/ArmPkg/Library/ArmGenericTimerPhyCounterLib/ArmGenericTimerPhyCounterLib.inf
+++ b/ArmPkg/Library/ArmGenericTimerPhyCounterLib/ArmGenericTimerPhyCounterLib.inf
@@ -22,7 +22,12 @@
 [Packages]
   MdePkg/MdePkg.dec
   ArmPkg/ArmPkg.dec
+  SiliciumPkg/SiliciumPkg.dec
 
 [LibraryClasses]
   ArmLib
   BaseLib
+  PcdLib
+
+[FixedPcd]
+  gSiliciumPkgTokenSpaceGuid.PcdTimerFreqOverwrite
diff --git a/ArmPkg/Library/ArmGenericTimerVirtCounterLib/ArmGenericTimerVirtCounterLib.c b/ArmPkg/Library/ArmGenericTimerVirtCounterLib/ArmGenericTimerVirtCounterLib.c
index 893125a111..4b1018b8dc 100644
--- a/ArmPkg/Library/ArmGenericTimerVirtCounterLib/ArmGenericTimerVirtCounterLib.c
+++ b/ArmPkg/Library/ArmGenericTimerVirtCounterLib/ArmGenericTimerVirtCounterLib.c
@@ -59,7 +59,14 @@ ArmGenericTimerGetTimerFreq (
   VOID
   )
 {
-  return ArmReadCntFrq ();
+  UINTN Freq;
+
+  Freq = ArmReadCntFrq ();
+  if (!Freq) {
+    Freq = FixedPcdGet64 (PcdTimerFreqOverwrite);
+  }
+
+  return Freq;
 }
 
 UINTN
diff --git a/ArmPkg/Library/ArmGenericTimerVirtCounterLib/ArmGenericTimerVirtCounterLib.inf b/ArmPkg/Library/ArmGenericTimerVirtCounterLib/ArmGenericTimerVirtCounterLib.inf
index 89fcb71c5f..5c29ece5a6 100644
--- a/ArmPkg/Library/ArmGenericTimerVirtCounterLib/ArmGenericTimerVirtCounterLib.inf
+++ b/ArmPkg/Library/ArmGenericTimerVirtCounterLib/ArmGenericTimerVirtCounterLib.inf
@@ -22,7 +22,12 @@
 [Packages]
   MdePkg/MdePkg.dec
   ArmPkg/ArmPkg.dec
+  SiliciumPkg/SiliciumPkg.dec
 
 [LibraryClasses]
   ArmLib
   BaseLib
+  PcdLib
+
+[FixedPcd]
+  gSiliciumPkgTokenSpaceGuid.PcdTimerFreqOverwrite
