diff --git a/MdeModulePkg/Bus/Usb/UsbBusDxe/UsbBus.h b/MdeModulePkg/Bus/Usb/UsbBusDxe/UsbBus.h
index 59d908d90d..bbc5b3280e 100644
--- a/MdeModulePkg/Bus/Usb/UsbBusDxe/UsbBus.h
+++ b/MdeModulePkg/Bus/Usb/UsbBusDxe/UsbBus.h
@@ -135,10 +135,10 @@ typedef struct _USB_HUB_API    USB_HUB_API;
 #define USB_CLEAR_FEATURE_REQUEST_TIMEOUT  10
 
 //
-// Bus raises TPL to TPL_NOTIFY to serialize all its operations
+// Bus raises TPL to TPL_CALLBACK to serialize all its operations
 // to protect shared data structures.
 //
-#define  USB_BUS_TPL  TPL_NOTIFY
+#define  USB_BUS_TPL  TPL_CALLBACK
 
 #define  USB_INTERFACE_SIGNATURE  SIGNATURE_32 ('U', 'S', 'B', 'I')
 #define  USB_BUS_SIGNATURE        SIGNATURE_32 ('U', 'S', 'B', 'B')
